Sync develop changes April 4 -April 22 to hdf5_1_14

lrknox opened the pull request.

bmribler committed:
Fixed dead links in docs (#4334)

bmribler committed:
Revert "Fixed dead links in docs (#4334)" (#4335)

This reverts commit 408fc6600292e945c8df32802eb9b9241093ce75.

hyoklee committed:
Fix grammar and content in HDF5Examples (#4333)

derobins committed:
Remove HD macros for C time functions (#4337)

* HDasctime
* HDclock
* HDctime
* HDdifftime
* HDgmtime
* HDlocaltime
* HDmktime
* HDtime
* HDtzset

HDgettimeofday will be done later

derobins committed:
Remove HD prefix from toupper/tolower (#4340)

bmribler committed:
Add VDS and SWMR to documentation (#4336)

derobins committed:
Clean up random number generator code (#4338)

* Clean up random number generator code

Depending on the platform, we use a mix of random, rand, and rand_r
to generate pseudo-random numbers, along with a messy set of ifdefs
in H5private.h. We are not a cryptographic library, only use random
numbers in our test code, and have no need for anything more than the
C standard's (s)rand(). There's no point dithering about using rand()
vs random() when we're also doing bad things like using mod to
restrict the range, which introduces bias.

Also removes CMake/configure checks for rand_r and random

* Remove random/rand_r checks from build system

* Fix missed HDrandom after GitHub merge

derobins committed:
Remove the getting started guide (#4339)

This was moved to the wiki

derobins committed:
Remove noise comment from H5DO.c (#4342)

derobins committed:
Add note about H5Iregister_type hash_table param (#4345)

The hash_table parameter is unused in HDF5 1.8.13 and later, when we moved from using hash tables to skip lists for storing IDs. The current hash table implementation does not use this parameter.

derobins committed:
Remove `#undef ERR` from H5private.h (#4346)

This is an old work-around that is no longer necessary.

The only place where we defined ERR is in h5import, and that code
has been updated (in this PR) to rename ERR to INVALID_TOKEN.

derobins committed:
Add NAMECHECK_ONLY to macro list in .clang-format (#4343)

This caused H5TS.c functions to be mis-formatted

derobins committed:
Remove HD from last non-win32 C89 functions (#4348)

Removes the HD prefix from the last C89 functions with no special
Windows equilvalent:

* rename
* rewind
* setbuf
* signal
* tmpfile

bmribler committed:
Fix dead links cont. (#4349)

Added img/images_to_copy.dox as a temporary solution because doxygen didn't copy
the images used in the examples/*.html files - will investigate more.  This was
necessary for the links to intro_SWMR.html and intro_VDS.html.

bmribler committed:
Fix a typo (#4352)

jhendersonHDF committed:
Fix use of possibly uninitialized memory in h5repack (#4354)

hyoklee committed:
Match content listing with content. (#4357)

byrnHDF committed:
Remove obsolete c++ doxygen config files (#4358)

hyoklee committed:
Fix typo VERSIO (#4363)

brtnfld committed:
Removed duplicate entries in compile line for h5* compiler wrappers (#4360)

* h5fc remove duplicate args
* h5 wrappers remove duplicate args

derobins committed:
Added CVE info to RELEASE.txt (#4367)

byrnHDF committed:
Change doxygen version from 1.10.0 to 1.9.7 (#4371)

byrnHDF committed:
Fix all doxygen versions (#4377)

hyoklee committed:
Change date to data in INSTALL_parallel (#4384)

hyoklee committed:
Fix a broken link for zlib url (#4383)

hyoklee committed:
Fix broken links and references in INSTALL_Auto.txt (#4382)

hyoklee committed:
Remove  signature in comment (#4366)

brtnfld committed:
switched from \returns to commonly used \return (#4389)

byrnHDF committed:
Add CXX flags (#4359)

derobins committed:
Remove TRACE macros (#4341)

jhendersonHDF committed:
Set available CMake build types for non-multi-config generators (#4330)

bmribler committed:
Correct the high-level library link (#4398)

* Correct the high-level library link

The link currently used points to 1.13 release.  Changed to use the md file instead.

* Update Doxyfile for md files

* Update to include md files

bmribler committed:
Add additional terms to the glossary (#4401)

Fixes GH #4390

brtnfld committed:
added missing doxygen \since (#4410)

hyoklee committed:
Update Intel oneAPI version to 2024.1 (#4412)

bmribler committed:
Update documentation (#4403)

- Added File Image Operations to Technical Notes

* Use a correct example location

byrnHDF committed:
Use aliases for URLs (#4409)

byrnHDF committed:
Fix CXX sanitizer by moving marco include to after CXX code. (#4414)

byrnHDF committed:
Update and use more URL alias (#4413)

bmribler committed:
Update documentation (#4415)

Added the replacement for https://docs.hdfgroup.org/archive/support/HDF5/doc/TechNotes/Version.html
Corrected some minor typos and another deadlink

byrnHDF committed:
Remove obsolete file (#4416)

hyoklee committed:
Update oneAPI version to 2024.1 for Daily Build (#4418)

bmribler committed:
Add training videos page (#4422)

lrknox committed:
Updates for release_docs/NEWSLETTER.txt. (#4423)

glennsong09 committed:
Add release process markdown file (#4312)

hyoklee committed:
Lowercase  in File Format Specification (#4424)

lrknox committed:
Fixing string in RELEASE_PROCESS.md that is causing a doxygen error - (#4430)

Unsupported xml/html tag <n> found.

derobins committed:
Allow clang-format to format H5Cpkg.h (#4429)

Co-authored-by: Larry Knox <lrknox@hdfgroup.org>

lrknox committed:
Fix broken links in md files needing full URLs.

lrknox commented:
I'm finding /develop/ in some paths to documentation.  Many of them should replaced with /v1_14/.  I'll be adding a new commit with those changes later this afternoon.

jhendersonHDF commented on doxygen/dox/ExamplesAPI.dox:
```diff
@@ -387,347 +387,345 @@ FORTRAN
 <td>
-<a href="https://github.com/HDFGroup/hdf5/blob/hdf5_1_14/HDF5Examples/C/H5T/h5ex_t_intatt.c">C</a>
-<a href="https://github.com/HDFGroup/hdf5/blob/hdf5_1_14/HDF5Examples/FORTRAN/H5T/h5ex_t_intatt_F03.F90">FORTRAN</a>
-<a href="https://github.com/HDFGroup/hdf5/blob/hdf5_1_14/HDF5Examples/JAVA/H5T/H5Ex_T_IntegerAttribute.java">Java</a>
```
Looks like this link to the Java example was dropped here rather than converted to the `\SRCURL` approach

lrknox replied:
added link in new commit, along with changes from /develop/ to /v1_14/ or /hdf5_1_14/ in some .md files.

lrknox committed:
Add URL for Java examples that was accidentally dropped.
Change paths from develop to 1.14 directories for some doxygen generated
files.

jhendersonHDF commented on CONTRIBUTING.md:
```diff
@@ -122,7 +122,7 @@ Please make sure that you check the items applicable to your pull request:
   * [ ] Is the pull request applicable to any other branches? If yes, which ones? Please document it in the GitHub issue.
   * [ ] Is the new code sufficiently documented for future maintenance?
-  * [ ] Does the new feature require a change to an existing API? See "API Compatibility Macros" document (https://docs.hdfgroup.org/hdf5/v1_14/api-compat-macros.html)
+  * [ ] Does the new feature require a change to an existing API? See "API Compatibility Macros" document (./api-compat-macros.html)
```
Does this one need to be changed back to a full link instead of a "./"?

lrknox replied:
Yes, fixed

jhendersonHDF commented on release_docs/RELEASE_PROCESS.md:
```diff
@@ -0,0 +1,163 @@
+    - Document any new known issues at the top of the list.
+3. Update the TESTED CONFIGURATION FEATURES SUMMARY in [RELEASE.txt](https://github.com/HDFGroup/hdf5/blob/hdf5_1_14/release_docs/RELEASE.txt) to correspond to features and options that have been tested during the maintenance period by the automated daily regression tests. 
+    - **See: Testing/Testing Systems(this is a page in confluence)**
+4. Update current compiler information for each platform in the PLATFORMS TESTED section of [RELEASE.txt](https://github.com/HDFGroup/hdf5/blob/hdff5_1_14/release_docs/RELEASE.txt).
```
Markdown checker failing here. `hdff5_1_14` -> `hdf5_1_14`

lrknox replied:
fixed

jhendersonHDF commented on release_docs/RELEASE_PROCESS.md:
```diff
@@ -0,0 +1,163 @@
+4. Update current compiler information for each platform in the PLATFORMS TESTED section of [RELEASE.txt](https://github.com/HDFGroup/hdf5/blob/hdff5_1_14/release_docs/RELEASE.txt).
+5. Review the [RELEASE.txt](https://github.com/HDFGroup/hdf5/blob/hdf5_1_14/release_docs/RELEASE.txt) for formatting and language to verify that it corresponds to guidelines found in **[Writing Notes in a RELEASE.txt(this is missing)]()** File.
+6. Review and update, if needed, the [README](https://github.com/HDFGroup/hdf5/blob/hdf5_1_14/README.md) and [COPYING](https://github.com/HDFGroup/hdf5/blob/hdf5_1_14/COPYING) files.
+7. Review and update all INSTALL_* files in [release_docs](https://github.com/HDFGroup/hdf5/tree/hdf5-1_14/release_docs), if needed.
```
Markdown checker failing here. I think `hdf5-1_14` -> `hdf5_1_14`?

lrknox replied:
yes, corrected

lrknox committed:
Correct typo.

lrknox committed:
Fix 2 more dead URLs.

lrknox committed:
Revert "Clean up random number generator code (#4338)"

This reverts commit 58e4d0842a9ab1c65722f672cabc38da22d48bd8.
